File: packages/rook_sdk_health_connect/lib/src/data/mapper/data_source_type_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_health_connect/src/data/proto/protos.pbenum.dart';

extension DataSourceTypeMappers on DataSourceType {
  DataSourceTypeProto toProto() {
    return switch (this) {
      DataSourceType.garmin => DataSourceTypeProto.GARMIN,
      DataSourceType.oura => DataSourceTypeProto.OURA,
      DataSourceType.polar => DataSourceTypeProto.POLAR,
      DataSourceType.fitbit => DataSourceTypeProto.FITBIT,
      DataSourceType.withings => DataSourceTypeProto.WITHINGS,
      DataSourceType.whoop => DataSourceTypeProto.WHOOP,
      _ => DataSourceTypeProto.WHOOP,
    };
  }
}

File: packages/rook_sdk_health_connect/android/src/main/kotlin/com/rookmotion/rook_sdk_health_connect/mapper/DataSourceTypeMappers.kt
package com.rookmotion.rook_sdk_health_connect.mapper

import com.rookmotion.rook.sdk.domain.enums.DataSourceType
import com.rookmotion.rook_sdk_health_connect.exception.UnknownDataSourceTypeException
import com.rookmotion.rook_sdk_health_connect.proto.DataSourceTypeProto

fun DataSourceTypeProto.toDataSourceType(): DataSourceType {
    return when (this) {
        DataSourceTypeProto.GARMIN -> DataSourceType.GARMIN
        DataSourceTypeProto.OURA -> DataSourceType.OURA
        DataSourceTypeProto.POLAR -> DataSourceType.POLAR
        DataSourceTypeProto.FITBIT -> DataSourceType.FITBIT
        DataSourceTypeProto.WITHINGS -> DataSourceType.WITHINGS
        DataSourceTypeProto.WHOOP -> DataSourceType.WHOOP
        else -> throw UnknownDataSourceTypeException()
    }
}
